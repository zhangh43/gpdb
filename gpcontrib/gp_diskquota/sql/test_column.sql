-- Test alter table add column
CREATE SCHEMA scolumn;
SELECT diskquota.set_schema_quota('scolumn', '1 MB');
SET search_path TO scolumn;
SELECT pg_sleep(5);

CREATE TABLE a2(i INT);
-- expect fail
INSERT INTO a2 SELECT generate_series(1,100000000);
-- expect fail
INSERT INTO a2 SELECT generate_series(1,10);
ALTER TABLE a2 ADD COLUMN j VARCHAR(50);
UPDATE a2 SET j = 'add value for column j';
SELECT pg_sleep(5);
-- expect insert failed after add column
INSERT INTO a2 SELECT generate_series(1,10);

DROP TABLE a2;
RESET search_path;
DROP SCHEMA scolumn;

