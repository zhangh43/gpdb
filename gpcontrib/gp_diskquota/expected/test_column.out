-- Test alter table add column
CREATE SCHEMA scolumn;
SELECT diskquota.set_schema_quota('scolumn', '1 MB');
 set_schema_quota 
------------------
 
(1 row)

SET search_path TO scolumn;
SELECT pg_sleep(5);
 pg_sleep 
----------
 
(1 row)

CREATE TABLE a2(i INT);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'i' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
-- expect fail
INSERT INTO a2 SELECT generate_series(1,100000000);
ERROR:  schema's disk space quota exceeded with name:scolumn
-- expect fail
INSERT INTO a2 SELECT generate_series(1,10);
ERROR:  schema's disk space quota exceeded with name:scolumn
ALTER TABLE a2 ADD COLUMN j VARCHAR(50);
UPDATE a2 SET j = 'add value for column j';
ERROR:  schema's disk space quota exceeded with name:scolumn
SELECT pg_sleep(5);
 pg_sleep 
----------
 
(1 row)

-- expect insert failed after add column
INSERT INTO a2 SELECT generate_series(1,10);
ERROR:  schema's disk space quota exceeded with name:scolumn
DROP TABLE a2;
RESET search_path;
DROP SCHEMA scolumn;
